tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,414 @@
+            raise ValueError(
+                f"The attribute: '{attribute_name}' is a primary key and can't be updated"
+            )
+
+    # Update attributes in dataset
+    for attribute_name in attributes:
+        attr_spec_dict = _make_spec_dict(
+            attribute_name=attribute_name, attribute_types=attribute_types
+        )
+        existing_attribute_spec = target_attribute_dict[attribute_name].spec()
+        new_type_class = attribute_type.from_json(attr_spec_dict["type"])
+        old_type_class = attribute_type.from_json(existing_attribute_spec.to_dict()["type"])
+
+        if new_type_class == old_type_class:
+            # Update description
+            if attribute_descriptions and attribute_name in attribute_descriptions.keys():
+                existing_attribute_spec = existing_attribute_spec.with_description(
+                    attribute_descriptions[attribute_name]
+                )
+                existing_attribute_spec.put()
+        elif override_existing_types:
+            # Update type
+            new_attr_spec = existing_attribute_spec.to_dict()
+            new_attr_spec["type"] = attr_spec_dict["type"]
+
+            # Update description
+            if attribute_descriptions and attribute_name in attribute_descriptions.keys():
+                new_attr_spec["description"] = attribute_descriptions[attribute_name]
+
+            # Remove and add attribute with new spec
+            target_dataset_attributes.delete_by_resource_id(
+                target_attribute_dict[attribute_name].resource_id
+            )
+            target_dataset_attributes.create(new_attr_spec)
+            LOGGER.info(f"Updated attribute '{attribute_name}' in {dataset_name}")
+        else:
+            LOGGER.info(
+                f"""The attribute '{attribute_name}' in {dataset_name} curently has
+                 the type '{str(old_type_class)}'. Set '{override_existing_types}' to
+                 True to update the type to '{str(new_type_class)}
+                """
+            )
+
+    return dataset
+
+
+def delete_attributes(*, dataset: Dataset, attributes: List[str] = None,) -> Dataset:
+    """Remove attributes from dataset by attribute name
+
+    Args:
+        dataset: An existing TUC dataset
+        attributes: list of attribute names to delete from dataset
+
+    Returns:
+        Updated Dataset
+
+    Raises:
+        requests.HTTPError: If any HTTP error is encountered
+        ValueError: {dataset_name} is not a source dataset
+        ValueError: attribute with {attribute_name} does not exist in {dataset_name}
+        ValueError: The attribute: '{attribute_name}' is a primary key and can't be removed
+        TypeError: attributes arg must be a List
+    """
+    dataset_name = dataset.name
+    if dataset.upstream_datasets():
+        raise ValueError(f"{dataset_name} is not a source dataset")
+
+    # Check input type is correct
+    if type(attributes) != list:
```

Comment:
Best practice is to use `isinstance`, which can handle subclasses whereas `type(<var>) == name` cannot:
```suggestion
    if not isinstance(attributes, list):
```

Reply:
I also don't think we should restrict this to being a list. What if someone passes a tuple, set, or other iterable?  Instead let's use `isinstance(t, Iterable)`, with `from typing import Iterable` (and updating the type-hint).